src/main/java/roomescape/infrastructure/LoginMemberArgumentResolver.java

```diff
@@ -0,0 +1,41 @@
+package roomescape.infrastructure;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.core.MethodParameter;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+import roomescape.member.LoginMember;
+import roomescape.member.Member;
+import roomescape.member.MemberService;
+
+@Component
+public class LoginMemberArgumentResolver implements HandlerMethodArgumentResolver {
+    private MemberService memberService;
+    private JwtTokenUtil jwtTokenUtil;
+
+    public LoginMemberArgumentResolver(MemberService memberService, JwtTokenUtil jwtTokenUtil) {
+        this.memberService = memberService;
+        this.jwtTokenUtil = jwtTokenUtil;
+    }
```

Comment:
`@Autowired` 어노테이션과 함께 생성자 주입을 하여 가독성을 좋게 할 수 있어요!

```suggestion
    @Autowired
    public LoginMemberArgumentResolver(MemberService memberService, JwtTokenUtil jwtTokenUtil) {
        this.memberService = memberService;
        this.jwtTokenUtil = jwtTokenUtil;
    }
```

Reply:
`@Autowired`로 변경하려고 하니 생성자를 통한 주입으로 변경을 스프링에서 추천하길래 잠깐 찾아봤어요! 공유해드리면 좋을 것 같아서 적어봅니다!

`@Autowired` 어노테이션을 사용하면 의존성 주입이 가능하지만, Spring에서는 생성자를 통한 의존성 주입을 권장합니다. 생성자 주입이 권장되는 이유는 여러 가지가 있습니다:

1. **불변성**: 생성자를 통해 주입된 의존성은 변경될 수 없습니다. 이는 객체의 상태를 더 안전하게 만듭니다.
2. **테스트 용이성**: 생성자 주입은 단위 테스트 작성 시 더 쉽고, Mockito와 같은 라이브러리로 주입할 의존성을 설정하기가 수월합니다.
3. **순환 참조 방지**: 생성자 주입은 컴파일 타임에 순환 의존성을 확인할 수 있도록 해주므로, 런타임에 발생할 수 있는 순환 참조 문제를 방지합니다.
4. **명확한 의존성**: 생성자를 통해 의존성을 주입하면, 해당 클래스가 어떤 의존성을 필요로 하는지 명확하게 알 수 있습니다.

<aside>
💡 Spring Boot 2.0 이후, `@Autowired`를 필드에 사용하는 것보다 생성자 주입을 더 권장하고 있습니다. 그래서 IDE나 코드 검사 도구가 생성자를 사용하라는 경고를 표시하는 것입니다.

</aside>